index.js

```diff
@@ -106,6 +106,7 @@ function asPromise(opts) {
 					const limitStatusCode = opts.followRedirect ? 299 : 399;
 
 					res.body = data;
+					res.requestUrl = opts.href || urlLib.resolve(urlLib.format(opts), opts.path);
```

Comment:
No, when sending an object we don't have `opts.href` and `urlLib.format(opts)` don't return the `opts.path` in it. I just put the `opts.href` because if it exist we don't need to send to the functions. But it is removable.